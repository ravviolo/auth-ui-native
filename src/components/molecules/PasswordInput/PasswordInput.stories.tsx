import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import { StoryScreen } from 'storybook/StoryScreen';

import { PasswordInput } from './PasswordInput';

storiesOf('PasswordInput', module)
  .addDecorator((getStory) => <StoryScreen>{getStory()}</StoryScreen>)
  .add('Default', () => <PasswordInput testID={text('testID', 'password-input-test-id')} />);
