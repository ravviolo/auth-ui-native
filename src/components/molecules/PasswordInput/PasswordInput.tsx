import { InputField, InputLabel, TextInputHandle } from 'components/atoms';
import { forwardRef } from 'react';
import { View } from 'react-native';

interface Props {
  testID: string;
}

export const PasswordInput = forwardRef<TextInputHandle, Props>(({ testID }, ref) => {
  return (
    <View testID={testID}>
      <InputLabel label="Password" testID="input-label-password-test-id" />
      <InputField ref={ref} testID="input-field-password-test-id" secureTextEntry />
    </View>
  );
});
