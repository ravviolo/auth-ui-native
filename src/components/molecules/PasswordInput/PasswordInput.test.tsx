import { render, screen, fireEvent, within } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { PasswordInput } from './PasswordInput';

import type { Props } from './PasswordInput';

describe('PasswordInput', () => {
  const props: Props = {
    testID: 'password-input-test-id',
  };

  it('should render input with text entry secured', () => {
    render(<PasswordInput {...props} />);

    const passwordInput = screen.getByTestId('password-input-test-id');
    const passwordInputField = within(passwordInput).getByTestId('input-field-password-test-id');

    expect(passwordInputField.props).toHaveProperty('secureTextEntry', true);
  });

  it('should render initially as empty input', () => {
    render(<PasswordInput {...props} />);

    const passwordInput = screen.getByTestId('password-input-test-id');
    const passwordInputField = within(passwordInput).getByTestId('input-field-password-test-id');

    expect(passwordInputField.props.value).toBe('');
  });

  it('should handle onChangeText events', () => {
    render(<PasswordInput {...props} />);

    const passwordInput = screen.getByTestId('password-input-test-id');
    const passwordInputField = within(passwordInput).getByTestId('input-field-password-test-id');

    fireEvent.changeText(passwordInputField, 'test input');

    expect(passwordInputField.props.value).toBe('test input');
  });

  it("should display 'Password' label", () => {
    render(<PasswordInput {...props} />);

    const passwordInput = screen.getByTestId('password-input-test-id');

    const passwordInputLabel = within(passwordInput).getByTestId('input-label-password-test-id');

    expect(passwordInputLabel).toHaveTextContent('Password');
  });

  it('should match snapshot', () => {
    const tree = testRenderer.create(<PasswordInput {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
