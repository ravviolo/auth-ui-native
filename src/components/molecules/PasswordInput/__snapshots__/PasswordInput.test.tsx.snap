// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordInput should match snapshot 1`] = `
<View
  testID="password-input-test-id"
>
  <Text
    style={
      Object {
        "color": "#3D3D3D",
        "marginBottom": 3,
      }
    }
    testID="input-label-password-test-id"
  >
    Password
  </Text>
  <TextInput
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    secureTextEntry={true}
    style={
      Array [
        Object {
          "borderColor": "#9E9E9E",
          "borderRadius": 8,
          "borderWidth": 1,
          "height": 40,
          "padding": 10,
        },
        Object {
          "borderColor": "#9E9E9E",
        },
      ]
    }
    testID="input-field-password-test-id"
    value=""
  />
</View>
`;
